### DISCUSSION TITLE
About parsing the GPIF xml in the GP file

### DISCUSSION THREAD

**Discussion by Jon-Millent** [CoderLine/alphaTab#693] [Q&A] [answered]
This is a great project, I am reading the code of this project, about parsing the GPIF xml in the GP file, I can't find the relevant documentation about the GPIF xml file on Google, do you know where there is a description about the GPIF xml format file Documentation website? Thank you so much!

**Comment by Danielku15** [comment, marked as answer]
Hi. 
Unfortunately there is no documentation about this. The Guitar Pro files are rather a proprietary format and the knowledge about it comes mostly from reverse engineering practices. From my knowledge Arobas Music only contributed twice some information about their format to the public. 

First time was with the [Guitar Pro 4 file format](http://dguitar.sourceforge.net/GP4format.html) from which the Guitar Pro 3 and 5 formats were further analyzed. But this was still the old file format. 

Next time was with Guitar Pro 6 with their proprietary GPX compression format [here](http://www.tuxguitar.com.ar/forum/3/1657/guitar-pro-6-gpx-files/) (the post of bSr43). 

In GP7 they dropped the proprietary GPX archive and just went for a Zip Archive with the files in. 

To really understand the score.gpif file, you need a bit experience on how the internal Guitar Pro data model looks like. The [alphaTab data model](https://alphatab.net/docs/reference/score) is quite aligned to the model of Guitar Pro so you could read our docs to get the basic understanding of some terminology and hierarchy.  

Then the GPIF is just basically just an XML hiearchy with this structure. The nesting is done by referencing the IDs of the particular elements and -1 usually means "empty". 

Starting from there, it's a matter of reading the XML, it is quite "developer friendly". From there, the reverse engineering starts in the way: Make tons of test files, and see what XML ends up in the file. With the base knowledge about the structure, music theory, Guitar Pro in general and maybe a bit midi, usually you can fit the puzzle pieces together.

**Reply by Jon-Millent** [reply]
Thank you very much for your reply. maybe it's a good choice for me to study music xml first. 😄
